File: src/main/java/org/aldo/api/service/interfaces/DailyHoursService.java
package org.aldo.api.service.interfaces;

import org.aldo.api.data.dto.CreateDailyHoursDto;
import org.aldo.api.data.dto.DailyDetailDto;
import org.aldo.api.data.dto.UpdateDailyHoursDto;
import org.springframework.security.core.Authentication;

import java.time.Month;
import java.time.Year;
import java.util.List;

public interface DailyHoursService {
    void createDailyHours(List<CreateDailyHoursDto> createDailyHoursDto, Authentication authentication);
    void updateDailyHours(List<UpdateDailyHoursDto> updateDailyHoursDto, Authentication authentication);
    List<DailyDetailDto> getWorkedHoursByProfessorIdAndMonthAndProjectCup(Month month, Year year, Integer professorId);
}

File: src/app/service/daily-hours/daily-hours.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { DailyHours } from '../../model/DailyHours';
import {DailyDetailDto} from "../../model/dto/DailyDetailDto";
import {UpdateDailyHoursDto} from "../../model/dto/UpdateDailyHoursDto";

@Injectable({
  providedIn: 'root'
})
export class DailyHoursService {
  private urlApi = 'http://localhost:8080/api/v1/collaborations/daily';

  constructor(private http: HttpClient) { }

  createDailyHours(dailyHours: DailyHours[]) {
    return this.http.post(this.urlApi + "/create", dailyHours);
  }

  updateDailyHours(updateDailyHours: UpdateDailyHoursDto[]) {
    return this.http.patch(
      this.urlApi + "/update",
      updateDailyHours,
      { params: {}}
    );
  }

  getDailyHours(month: string, year: number, professorId: number){
    return this.http.get<DailyDetailDto[]>(
      this.urlApi,
      { params: {
        month: month.toString(),
        year: year.toString(),
        professor: professorId.toString()
      }
    });
  }
}
